app/src/main/java/com/hedvig/app/feature/keygear/KeyGearValuationActivity.kt

```diff
@@ -44,31 +82,89 @@ class KeyGearValuationActivity : BaseActivity(R.layout.activity_key_gear_valuati
             onBackPressed()
         }
 
-        continueButton.setHapticClickListener {
-            val price = priceInput.getText()
+        save.setHapticClickListener {
+            if (isUploading) {
+                return@setHapticClickListener
+            }
+            isUploading = true
+            transitionToUploading()
 
+            val price = priceInput.getText()
             safeLet(date, id) { date, id ->
                 val monetaryValue =
                     MonetaryAmountV2Input.builder().amount(price).currency("SEK").build()
+
                 model.updatePurchaseDateAndPrice(id, date, monetaryValue)
-                //TODO
-                onBackPressed()
+
             }
         }
 
         priceInput.setOnChangeListener {
             val text = priceInput.getText()
             setButtonState(text.isNotEmpty(), date != null)
         }
+
+        model.finishedUploading.observe(this) { finishedUploading ->
+            finishedUploading?.let {
+                if (finishedUploading) {
+                    id?.let { id ->
+                        startActivity(
+                            KeyGearValuationInfoActivity.newInstance(
+                                this,
+                                id
+                            )
+                        )
+                        finish()
+                    }
+                }
+            }
+        }
+    }
+
+    private fun transitionToUploading() {
+        loadingIndicator.show()
+        val startCornerRadius = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            ((saveContainer.background as RippleDrawable).getDrawable(0) as GradientDrawable).cornerRadius
+        } else {
+            BUTTON_CORNER_RADIUS
+        }
+        ValueAnimator.ofInt(saveContainer.width, saveContainer.height).apply {
+            interpolator = AccelerateDecelerateInterpolator()
+            duration = SAVE_BUTTON_TRANSITION_DURATION
+            addUpdateListener { va ->
+                saveContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+                    width = va.animatedValue as Int
+                }
+                save.alpha = 1 - va.animatedFraction
+                loadingIndicator.alpha = va.animatedFraction
+                val backgroundShape =
+                    ((saveContainer.background as? RippleDrawable)?.getDrawable(0) as? GradientDrawable)?.mutate() as? GradientDrawable
+                backgroundShape?.cornerRadius =
+                    boundedLerp(startCornerRadius, saveContainer.height / 2f, va.animatedFraction)
+            }
+            start()
+        }
     }
 
     private fun setButtonState(hasPrice: Boolean, hasDate: Boolean) {
-        continueButton.isEnabled = hasPrice && hasDate
+        if (hasPrice && hasDate) {
+            save.isEnabled = true
+            saveContainer.backgroundTintList =
```

Comment:
Should we use the disabled-feature here as well? 🙂

Reply:
Dont know if that is possible since i disable the button but change the color of the parent frameLayout

Reply:
Is it possible to set the FrameLayout to be disabled? If not, we can disregard it, but it's interesting to see if we can solve the problem using the existing facilities, so to speak